core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/weaver/corda/app/interop/flows/AssetTransferFlows.kt

```diff
@@ -0,0 +1,859 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.weaver.corda.app.interop.flows
+
+import arrow.core.*
+import co.paralleluniverse.fibers.Suspendable
+import com.weaver.corda.app.interop.contracts.AssetTransferContract
+import com.weaver.corda.app.interop.states.AssetPledgeState
+import com.weaver.corda.app.interop.states.AssetClaimStatusState
+import com.weaver.corda.app.interop.states.NetworkIdState
+import com.weaver.protos.common.asset_transfer.AssetTransfer
+import com.weaver.protos.corda.ViewDataOuterClass
+import com.google.protobuf.ByteString
+
+import net.corda.core.contracts.ContractState
+import net.corda.core.contracts.CommandData
+import net.corda.core.contracts.Command
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.contracts.TimeWindow
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.contracts.ReferencedStateAndRef
+import net.corda.core.contracts.requireThat
+import net.corda.core.contracts.StaticPointer
+
+import net.corda.core.identity.Party
+
+import net.corda.core.flows.*
+import net.corda.core.node.services.queryBy
+import net.corda.core.node.services.vault.QueryCriteria
+import net.corda.core.node.ServiceHub
+import net.corda.core.node.StatesToRecord
+import net.corda.core.transactions.TransactionBuilder
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.LedgerTransaction
+import net.corda.core.utilities.OpaqueBytes
+import net.corda.core.utilities.unwrap
+import net.corda.core.serialization.CordaSerializable
+import java.time.Instant
+import java.util.Base64
+import java.util.Calendar
+
+/**
+ * Enum for communicating the role of the responder from initiator flow to responder flow.
+ */
+@CordaSerializable
+enum class AssetTransferResponderRole {
+    LOCKER, ISSUER, OBSERVER
```

Comment:
Can we re-use this enum class from the one declared in HTLC. It seems exactly same as there.
Or else if you want to keep two, then LOCKER can be changed to PLEDGER, just to show its different from the other.

Reply:
I changed LOCKEER to PLEDGER, Sandeep.